src/index.ts

```diff
@@ -1,60 +1,100 @@
-import * as core from '@actions/core';
-import * as exec from '@actions/exec';
-import * as io from '@actions/io';
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+import * as io from "@actions/io";
 import * as fs from "fs";
 import * as path from "path";
 
-export async function run(): Promise<void> {
-    const imageInput = core.getInput('image', { required: true });
-    const tag = core.getInput('tag') || 'latest';
-    const registry = core.getInput('registry', { required: true });
-    const username = core.getInput('username', { required: true });
-    const password = core.getInput('password', { required: true });
-    const tlsVerify = core.getInput('tls-verify');
-    const digestFileInput = core.getInput('digestfile');
+interface Response {
```

Comment:
Can you give this a better name? It's the result of an `exec`, so you could call it `ExecResult`.